File: ios/Classes/IrisMethodChannelPlugin.m
#import "IrisMethodChannelPlugin.h"
#include "../../src/iris_event.h"

@implementation IrisMethodChannelPlugin
+ (void)registerWithRegistrar:(NSObject<FlutterPluginRegistrar>*)registrar {
  FlutterMethodChannel* channel = [FlutterMethodChannel
      methodChannelWithName:@"iris_method_channel"
            binaryMessenger:[registrar messenger]];
  IrisMethodChannelPlugin* instance = [[IrisMethodChannelPlugin alloc] init];
  [registrar addMethodCallDelegate:instance channel:channel];
}

- (void)handleMethodCall:(FlutterMethodCall*)call result:(FlutterResult)result {
  if ([@"getPlatformVersion" isEqualToString:call.method]) {
    result([@"iOS " stringByAppendingString:[[UIDevice currentDevice] systemVersion]]);
  } else {
    result(FlutterMethodNotImplemented);
  }
}

/// dummy function to avoid symbols striping when building static library.
+ (void)_irisMethodChannelDummyFunc {
  EventParam p;
	InitDartApiDL(NULL);
  Dispose();
  OnEvent(NULL);
  RegisterDartPort(0);
  UnregisterDartPort(0);
}

@end

File: macos/iris_method_channel/Sources/iris_method_channel/IrisMethodChannelPlugin.swift
import Cocoa
import FlutterMacOS

#if canImport(iris_method_channel_objc)
  import iris_method_channel_objc
#endif

public class IrisMethodChannelPlugin: NSObject, FlutterPlugin {
  public static func register(with registrar: FlutterPluginRegistrar) {
    let channel = FlutterMethodChannel(name: "iris_method_channel", binaryMessenger: registrar.messenger)
    let instance = IrisMethodChannelPlugin()
    registrar.addMethodCallDelegate(instance, channel: channel)
  }

  public func handle(_ call: FlutterMethodCall, result: @escaping FlutterResult) {
    switch call.method {
    case "getPlatformVersion":
      result("macOS " + ProcessInfo.processInfo.operatingSystemVersionString)
    default:
      result(FlutterMethodNotImplemented)
    }
  }
}
